xbrowse_server/phenotips/utilities.py

```
@@ -0,0 +1,121 @@
+  admin_uname='Admin'
+  admin_pwd='admin'
+  headers={"Content-Type": "application/x-www-form-urlencoded"}
+  data={'parent':'XWiki.XWikiUsers'}
+  url = settings.PHENOPTIPS_HOST_NAME + '/rest/wikis/xwiki/spaces/XWiki/pages/' + new_user_name
+  do_authenticated_PUT(admin_uname,admin_pwd,url,data,headers) 
+  data={'className':'XWiki.XWikiUsers',
+        'property#first_name':new_user_first_name,
+        'property#last_name':new_user_last_name,
+        'property#email':email_address,
+        'property#password':new_user_pwd
+        }
+  url=settings.PHENOPTIPS_HOST_NAME + '/rest/wikis/xwiki/spaces/XWiki/pages/' + new_user_name + '/objects'
+  do_authenticated_POST(admin_uname,admin_pwd,url,data,headers)
+  
+  
+#adds a collaborator to an existing patient. Requires an existing collaborator username, patient_eid (PXXXX..)
+def add_read_only_collaborator_phenotips_patient(collaborator_username,patient_eid):
+  #we need to put this password in a non-checkin file
+  admin_uname='Admin'
+  admin_pwd='admin'
+  headers={"Content-Type": "application/x-www-form-urlencoded"}
+  data={'collaborator':'XWiki.' + collaborator_username,
+        'patient':patient_eid,
+        'accessLevel':'view',
+        'xaction':'update',
+        'submit':'Update'}
+  url = settings.PHENOPTIPS_HOST_NAME + '/bin/get/PhenoTips/PatientAccessRightsManagement?outputSyntax=plain'
+  do_authenticated_POST(admin_uname,admin_pwd,url,data,headers)
+  
+  
+#do a PUT call to phenotips
+def do_authenticated_PUT(uname,pwd,url,data,headers):
+  try:
+    request=requests.put(url,data=data,auth=(uname,pwd),headers=headers)
+  except Exception as e:
+    print 'error in do_authenticated_PUT:',e,
+    return e
+  
+#do a PUT call to phenotips
+def do_authenticated_POST(uname,pwd,url,data,headers):
+  try:
+    request=requests.post(url,data=data,auth=(uname,pwd),headers=headers)
+  except Exception as e:
+    print 'error in do_authenticated_POST:',e,
```

for error handling inside moduels, I feel we should stick with python logging through-out (for consistency and so we can adjust format in one location) and also - for errors inside utilities functions, etc. we should raise exceptions instead of returning values to reduce the chances of silently ignoring errors and to clarify error handling code